qm-plugins/qm-cron/class-qm-cron-collector.php

```diff
@@ -0,0 +1,168 @@
+		'wp_update_plugins',
+		'wp_update_themes',
+		'wp_version_check',
+	];
+
+	/**
+	 * @return string
+	 */
+	public function name() {
+		return esc_html__( 'Cron', 'query-monitor' );
+	}
+
+	/**
+	 * @return void
+	 */
+	public function process() {
+		$this->process_crons();
+		$this->process_schedules();
+	}
+
+	/**
+	 * Processes the cron jobs.
+	 *
+	 * @return void
+	 */
+	private function process_crons() {
+		$this->data['doing_cron'] = get_transient( 'doing_cron' ) ? true : false;
+
+		$crons = _get_cron_array();
+		if ( is_array( $crons ) && ! empty( $crons ) ) {
+			$this->data['crons'] = $crons;
+			$this->sort_count_crons( $crons );
+			$this->process_next_event_time( $crons );
+		}
+	}
+
+	/**
+	 * Sort and count crons.
+	 *
+	 * This function sorts the cron jobs into core crons, and custom crons. It also tallies
+	 * a total count for the crons as this number is otherwise tough to get.
+	 *
+	 * @param array $crons Cron events
+	 */
+	private function sort_count_crons( $crons ) {
+		$total_crons      = 0;
+		$total_core_crons = 0;
+		$total_user_crons = 0;
+		$core_crons       = [];
+		$user_crons       = [];
+		foreach ( $crons as $time => $time_cron_array ) {
+			foreach ( $time_cron_array as $hook => $data ) {
+				$total_crons += count( $data );
+
+				if ( in_array( $hook, $this->core_cron_hooks, true ) ) {
+					$core_crons[ $time ][ $hook ] = $data;
+					$total_core_crons            += count( $data );
+				} else {
+					$user_crons[ $time ][ $hook ] = $data;
+					$total_user_crons            += count( $data );
+				}
+			}
+		}
+
+		$this->data['total_crons']      = $total_crons;
+		$this->data['total_core_crons'] = $total_core_crons;
+		$this->data['core_crons']       = $core_crons;
+		$this->data['user_crons']       = $user_crons;
+		$this->data['total_user_crons'] = $total_user_crons;
+	}
+
+	/**
+	 * Process next cron event time
+	 *
+	 * @param array Cron events
+	 */
+	private function process_next_event_time( $crons ) {
+		$cron_times          = array_keys( $crons );
+		$unix_time_next_cron = (int) $cron_times[0];
+		$time_next_cron      = gmdate( 'Y-m-d H:i:s', $unix_time_next_cron );
+
+		$this->data['next_event_time']['human_time'] = $time_next_cron;
+		$this->data['next_event_time']['unix']       = $unix_time_next_cron;
+	}
+
+	/**
+	 * Sorting method for cron schedules. Order by schedules interval.
+	 *
+	 * @param array $schedule_a First element of comparison pair.
+	 * @param array $schedule_b Second element of comparison pair.
+	 *
+	 * @return int Return 1 if $schedule_a argument 'interval' greater then $schedule_b argument 'interval',
+	 *             0 if both intervals equivalent and -1 otherwise.
+	 */
+	private function schedules_sorting( $schedule_a, $schedule_b ) {
+		if ( (int) $schedule_a['interval'] === (int) $schedule_b['interval'] ) {
+			return 0;
+		} else {
+			return ( ( (int) $schedule_a['interval'] > (int) $schedule_b['interval'] ) ? 1 : -1 );
+		}
```

Comment:
What do you think about this:

```suggestion
		return (int) $schedule_a['interval'] <=> (int) $schedule_b['interval'];
```

RefL [spaceship operator](https://www.php.net/manual/en/migration70.new-features.php#migration70.new-features.spaceship-op)